cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml

```diff
@@ -37,6 +37,10 @@ spec:
           readOnly: true
         - name: config-volume
           mountPath: /etc/google-fluentd/config.d
+        args:
```

Comment:
NOTE: `STACKDRIVER_METADATA_AGENT_URL` is set up as an environment variable regardless whether we are ingesting logs via old resources (in which case `STACKDRIVER_METADATA_AGENT_URL` is not needed) or new resources. Just because there seems to be no need to distinguish the case.

Thoughts?

Reply:
I'm ok with setting this unconditionally, however, why aren't we setting this as a normal environment variable? Injecting it at this point is actually problematic as it harms the ability to shut down cleanly.

```
        - env:
          - name: STACKDRIVER_METADATA_AGENT_URL
            value: {{ stackdriver_metadata_agent_url }}
```

Is what I'm suggesting.

Reply:
Good point. Changed.